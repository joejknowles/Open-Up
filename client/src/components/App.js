import React, { Component } from 'react';
import '../styles/App.css';
import DayBooking from './dayBooking';
import Errors from './alerts/errors';

import { Provider } from 'react-redux';

class App extends Component {
  render() {
    return (
      <Provider store={ this.props.store }>
        <div className="App">
          <Errors />
          <DayBooking />
        </div>
      </Provider>
    );
  }
}

export default App;
