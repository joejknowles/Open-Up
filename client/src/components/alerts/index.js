import React from 'react';
import { connect } from 'react-redux';
import ReactCSSTransitionsGroup from 'react-addons-css-transition-group';

import '../../styles/Alert.css';

import { alertsSelector, createAlertSelector } from '../../reducers';
import { removeAlert } from '../../actions/alerts'

export const Alert = ({ message, onClick, type }) => (
  <div
    className={ `Alert ${ type === 'ERROR' ? 'Error' : 'Success' }`}
    { ...{ onClick } }>
    { message }
  </div>
);

const mapDispatchToProps = (dispatch, { id }) => ({
  onClick: () => (dispatch(removeAlert(id)))
});
const mapStateToProps = (state, { id }) => createAlertSelector(id);

const ConnectedError = connect(
  mapStateToProps,
  mapDispatchToProps
)(Alert);

export const Alerts = ({ alerts }) => (
  <div className="Alerts">
    <ReactCSSTransitionsGroup
      transitionName="alert"
      transitionEnterTimeout={500}
      transitionLeaveTimeout={400}
      >
      {
        alerts.map((id) => (
          <ConnectedError key={ id } { ...{ id } } />
        ))
      }
    </ReactCSSTransitionsGroup>
  </div>
);

export default connect((state) => (
  { alerts: alertsSelector(state) }
))(Alerts);
