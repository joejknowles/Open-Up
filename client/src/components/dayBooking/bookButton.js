import React from 'react';
import { connect } from 'react-redux';
import format from 'date-fns/format'
import { createSlotSelector } from '../../reducers'

export const BookButton = ({ startTime, endTime}) => (
  <button>book { formatTime(startTime) } to { formatTime(endTime) }</button>
);

const formatTime = (time) => (
  format(time, 'H:mm')
);

const mapStateToProps = (state, { id }) => (
  createSlotSelector(id)
);

export default connect(mapStateToProps)(BookButton);
