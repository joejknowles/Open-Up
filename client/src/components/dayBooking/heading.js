import React from 'react';
import { connect } from 'react-redux';
import { selectedDateSelector } from '../../reducers';
import format from 'date-fns/format'
import { nextDay } from '../../actions/changeDay';

const dateTitle = (date) => (
  `Today ${format(date, 'D/M/YYYY')}`
);

export const Heading = ({ date, onNextDayClick }) => (
  <div className="Heading">
    <h3>{ dateTitle(date) }</h3>
    <a
      className="next-day"
      onClick={ onNextDayClick }>
      Next Day
    </a>
  </div>
);

export const mapStateToProps = (state) => ({
  date: selectedDateSelector(state)
});

const mapDispatchToProps = (dispatch) => ({ onNextDayClick: () => dispatch(nextDay) });

export default connect(mapStateToProps, mapDispatchToProps)(Heading);
