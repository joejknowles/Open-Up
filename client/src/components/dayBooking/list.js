import React from 'react';
import '../../styles/DayBooking.css';
import { connect } from 'react-redux';
import { slotsSelector } from '../../reducers';
import Slot from './slot'

export const List = ({ slots = [] }) => (
  slots.length === 0 ?
  <p className="message">No slots available</p> :
  (<div className="List">
    { slots.map(( id ) => (
      <Slot key={ id } { ...{ id } } />
    )) }
  </div>)
);

const mapStateToProps = (state) => ({
  slots: slotsSelector(state)
});

const connectedList = connect(mapStateToProps)(List);

export default connectedList;
