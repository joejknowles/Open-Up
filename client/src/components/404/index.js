import React from 'react';
import { Link } from 'react-router';
import '../../styles/App.css';

export default () => (
  <div className="App">
    <h3>Page not found</h3>
    <Link to="venue">Home</Link>
  </div>
);
