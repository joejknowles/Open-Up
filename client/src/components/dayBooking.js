import React, { Component } from 'react';
import '../styles/App.css';
import { connect } from 'react-redux';

export class DayBooking extends Component {
  render() {
    const date = this.props.date ?
      this.props.date.toLocaleDateString("en-GB") :
      '';
    const slots = this.props.slots || []
    return (
      <div>
        <h3>Today { date }</h3>
        { slots.map(({ id }) => (
            <button key={ id } ></button>
        )) }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  slots: state.allSlots.map((id) => state.slotsById[id]),
  date: state.selectedDate
});

export default connect(mapStateToProps)(DayBooking);
