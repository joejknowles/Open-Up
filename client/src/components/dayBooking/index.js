import React, { Component } from 'react';
import '../../styles/App.css';
import { connect } from 'react-redux';
import { fetchSlots } from '../../actions';

import LoadingIndicator from '../loadingIndicator';
import Heading from './heading';
import List from './list';

export class DayBooking extends Component {
  componentDidMount() {
    this.props.fetchSlots();
  }

  render() {
    if (this.props.isLoading) return (<LoadingIndicator />);
    return (
      <div>
        <Heading />
        <List slots={ this.props.slots } />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  slots: state.allSlots.map((id) => state.slotsById[id]),
  date: state.selectedDate,
  isLoading: state.isLoading
});

const mapDispatchToProps = (dispatch) => ({
  fetchSlots: fetchSlots(dispatch)
});

const mergeProps = ({ slots, isLoading, date }, dispatchProps) => ({
  slots, isLoading,
  fetchSlots: dispatchProps.fetchSlots(date)
});

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(DayBooking);
