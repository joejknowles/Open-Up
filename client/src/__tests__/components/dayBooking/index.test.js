import { DayBooking } from '../../../components/dayBooking';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

import { shallow } from 'enzyme';
const shallowRenderer = ReactTestUtils.createRenderer();

it('matches snapshot', () => {
  const tree = shallowRenderer.render(
    <DayBooking />
  );
  expect(tree).toMatchSnapshot();
});

it('updates slots on didMount', () => {
  const fetchSlots = jest.fn();
  const wrapper = shallow(
    <DayBooking
      { ...{ fetchSlots } }
      />
  );
  wrapper.instance().componentDidMount();
  expect(fetchSlots.mock.calls.length).toBe(1);
});

it('updates slots on didUpdate when date changes', () => {
  const fetchSlots = jest.fn();
  const wrapper = shallow(
    <DayBooking { ...{ fetchSlots } } date={ new Date(1)} />,
    { lifecycleExperimental: true }
  );
  wrapper.setProps({date: new Date(10000000000)});
  expect(fetchSlots.mock.calls.length).toBe(2);
});

it("doesn't update slots on didUpdate when date doesn't change", () => {
  const fetchSlots = jest.fn();
  const wrapper = shallow(
    <DayBooking { ...{ fetchSlots } } date={ new Date(10000000000)} test={ 'right test '} />,
    { lifecycleExperimental: true }
  );
  wrapper.setProps({
    otherProp: 'something irrelevant',
    date: new Date(10000000000)});
  expect(fetchSlots.mock.calls.length).toBe(1);
});
