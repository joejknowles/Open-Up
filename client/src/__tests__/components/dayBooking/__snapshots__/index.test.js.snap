exports[`test matches snapshot 1`] = `
<div>
  <Connect(Component) />
  <Connect(Component)>
    <Connect(Component) />
  </Connect(Component)>
</div>
`;
